src/Shared/DefaultSerializationContext.cs

```diff
@@ -16,17 +16,142 @@
 
 #endregion
 
+using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Runtime.CompilerServices;
 using Grpc.Core;
 
 namespace Grpc.Shared
 {
     internal sealed class DefaultSerializationContext : SerializationContext
     {
-        public byte[]? Payload { get; set; }
+        private byte[]? _array;
+        private InternalState _state;
+
+        public bool DirectSerializatedSupported { get; internal set; }
+        public int? PayloadLength { get; set; }
+        public IBufferWriter<byte>? BufferWriter { get; internal set; }
+
+        private enum InternalState
```

Comment:
Curious if we should share the states with HttpContextSerializationContext.cs? These look the same.

Reply:
I don't think it is worth sharing one enum.